crates/services/txpool/src/txpool.rs

```diff
@@ -126,12 +126,22 @@ impl<ViewProvider, WasmChecker> TxPool<ViewProvider, WasmChecker> {
     }
 
     /// Return all sorted transactions that are includable in next block.
-    pub fn sorted_includable(&self) -> impl Iterator<Item = ArcPoolTx> + '_ {
+    pub fn sorted_includable(
+        &self,
+        max_transactions: u16,
+    ) -> impl Iterator<Item = ArcPoolTx> + '_ {
         self.by_ratio_gas_tip
             .sort
             .iter()
             .rev()
-            .map(|(_, tx)| tx.clone())
+            .enumerate()
+            .map_while(move |(i, (_, tx))| {
+                if i <= (max_transactions as usize) {
```

Comment:
IIUC, here I need to check against `max_transactions - 1` to allow for the mint transaction to be included last. Probably need to consider also the slots reserved for FTI transactions (is the mint transaction considered to be a FTI transaction?)